include/pybind11/pybind11.h

```diff
@@ -222,7 +222,7 @@ class cpp_function : public function {
 
         /* Dispatch code which converts function arguments and performs the actual function call */
         rec->impl = [](function_call &call) -> handle {
-            cast_in args_converter;
+            cast_in args_converter{};
```

Comment:
```suggestion
            cast_in args_converter;
```

Just testing to see if this is causing the 3.12 failures.